Build failure due to implicit declaration of 'static_assert'

Unable to build blemesh_light target on latest code (0-dev).

**Compilation Error**
`
Error: repos/apache-mynewt-core/kernel/os/src/os_mempool.c: In function 'os_memblock_from':
repos/apache-mynewt-core/kernel/os/src/os_mempool.c:308:5: error: implicit declaration of function 'static_assert' [-Werror=implicit-function-declaration]
     static_assert(sizeof block_addr == sizeof baddr32,
     ^
cc1: all warnings being treated as errors
`


**Target YML:**
`
target.app: "@apache-mynewt-nimble/apps/blemesh_light"
target.bsp: "@apache-mynewt-core/hw/bsp/nordic_pca10056"
target.build_profile: debug
`

**Syscfg YML:**
`
syscfg.vals:
    BLE_MESH_DEV_UUID: "(uint8_t[16]){0x22, 0x20, 0}"
    BLE_MESH_PB_GATT: 1
`

**Package YML:**
`
pkg.name: targets/blemesh_light
pkg.type: target
pkg.description: 
pkg.author: 
pkg.homepage: 
`

--- a/libc/baselibc/include/assert.h
+++ b/libc/baselibc/include/assert.h
@@ -27,7 +27,7 @@ extern "C" {
 
 #endif
 
-#if __STDC_VERSION__ >= 201112L && !defined __cplusplus
+#if !defined __cplusplus
 #define static_assert _Static_assert
 #endif
 

